app/services/concerns/xccdf/rule_references.rb

```diff
@@ -7,7 +7,9 @@ module RuleReferences
 
     included do
       def save_rule_references
-        @rule_references ||= @op_rule_references.map do |op_reference|
+        @rule_references ||= RuleReference.new_from_openscap_parser(
+          @op_rule_references
+        ).map do |op_reference|
           ::RuleReference.from_openscap_parser(op_reference)
```

Comment:
IMO `from_openscap_parser` could be for 1 record - `all_from_openscap_parser` could be the same thing but accept a collection of op objects (and call `from_openscap_parser`) and return AR objects.